### DISCUSSION TITLE
[EMSUSD-2138] correct way to use exportMaterialCollections

### DISCUSSION THREAD

**Discussion by willDTF** [Autodesk/maya-usd#4020] [Q&A]
I need to export USD from maya with material collection assigned to objects

reading usd-maya [documentation](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/commands/Readme.md#command-flags-1), I need exportMaterialCollections and according to this [ticket resolved](https://github.com/Autodesk/maya-usd/pull/2442), i dont need to setup materialCollectionsPath argument. so i tried :

`cmds.mayaUSDExport( exportMaterialCollections=True, file=usdFilePath, )`

this doesn't work, no collection are created. then :

`cmds.mayaUSDExport( materialCollectionsPath='/Mtl', exportMaterialCollections=True, file=usdFilePath, )`

same result. i tried many variations, without success. The only step forward i've made is with :

`cmds.mayaUSDExport( exportCollectionBasedBindings=True, exportMaterialCollections=True, file=usdFilePath, ) `
material collections is created with the addition of exportCollectionBasedBindings, but no objects are in the collections

`cmds.mayaUSDExport( exportCollectionBasedBindings=True, exportMaterialCollections=True, file=usdFilePath,materialCollectionsPath='/Mtl' )`
 gives the same result that the previous, collections exist but empty

How can I properly export a USD file with material collections assigned ?

thank you

**Comment by wallworm** [comment]
Sorry I missed this post. We'll investigate and get back to you.

**Reply by willDTF** [reply]
Hello, I’ve made some progress. Updating to maya-usd last version make it work.
I have a little question, can we force it to be in a subdirectory ? For now it works only if materialCollectionsPath is set to the root Xform

<img width="279" alt="image" src="https://github.com/user-attachments/assets/40076d05-c18a-4da0-b6a2-e907dbaac008" />

Using ‘/TOM’ as path - Collections exist, objects are members
using ‘/TOM/mtl’ as path - Collections exist, but no objects are members of the collections

Hierarchy is the same in both case

**Reply by wallworm** [reply]
Thank you for the suggestion. We will look into this.

**Reply by santosg87** [reply]
@willDTF Not sure if we are missing something, or if this is just a matter of versioning, but testing the workflow on my side with our internal builds, i am seeing collection members as expected, as far as I can tell.

creating a simple Maya scene to mimic yours, then exporting:
```
import maya.cmds as cmds
usdFilePath = r'''C:\tmp\emsusd-2184-deep-mat-path\tom_new.usda'''
cmds.mayaUSDExport(
    exportCollectionBasedBindings=True,
    exportMaterialCollections=True,
    defaultPrim='TOM',
    rootPrim='TOM',
    rootPrimType='xform',
    file=usdFilePath,
    materialCollectionsPath='/TOM/mtl')
```

results in a file usd file, with the collections as i would expect:
```
    def Scope "mtl" (
        prepend apiSchemas = ["CollectionAPI:material:standardSurface2SG", "CollectionAPI:material:blinn1SG", "CollectionAPI:material:standardSurface3SG"]
    )
    {
        rel collection:material:blinn1SG:includes = </TOM/pCone2>
        rel collection:material:standardSurface2SG:includes = </TOM/pCone1>
        rel collection:material:standardSurface3SG:includes = </TOM/pCone3>
```

is this not the result you are expecting to see?
